Clean flasher debug output and reduce delays

Gravemind opened the pull request:
At first I tried to fix the few `\n` that changed since the flasher debug output refactor, but I ended up replacing `debug_msg` and `print_result` by a new `debug_printf` and `debug_result`.

The changes in the output are minor (a `\n` removed, one added, a few words changed/added ...), so I let you decide if the code changes are worth it.

On top of that commit I also did a separate one to reduce the setup and loop delay from 5 seconds to 2 seconds. So, it now delays only 4 seconds before flashing (instead of 10 seconds), seems more reasonable, especially because I flash with `make -B flash && sleep 2 && stty -F /dev/ttyACM0 9600 cs8 && cat /dev/ttyACM0`.

I did a dozen of flashes with this code while working on keyscanner without any issue.

obra commented:
I appreciate the cleanup, but the overhead of this one is a little bit heavy.
I've been working toward being able to integrate the flashing tool into a regular keyboard firmware.

This change takes us from:
Sketch uses 10468 bytes (36%) of program storage space. Maximum is 28672 bytes.
Global variables use 291 bytes (11%) of dynamic memory, leaving 2269 bytes for local variables. Maximum is 2560 bytes.
to
Sketch uses 12116 bytes (42%) of program storage space. Maximum is 28672 bytes.
Global variables use 303 bytes (11%) of dynamic memory, leaving 2257 bytes for local variables. Maximum is 2560 bytes.

The RAM usage doesn't bother me, but the extra 1.5k of flash is a heavy price to pay :/
Do you think it's possible to get that down while retaining what is otherwise a really nice improvement?

Gravemind commented:
I could not get any call call to `printf` without adding at least 1KB of prog mem, so I went back to using `Serial.print` (at first to compare, then commit).

Now, compared to f/configurable-scanners-and-debouncers:
- renamed `debug_msg` and `print_result` to `debug_print` and `debug_resultln`
- tweaked a couple of `\n` and messages
- removed the `PSTR()/F()` macros, so it uses ~40B less prog mem usage but **adds ~500B of RAM** (**not sure if that's a good trade-off**, I can put them back)
- reduced delay

obra commented:
Thanks!

Since we use progmem features elsewhere in the core, I suspect we're
already paying that penalty, so I'd probably error on the side of reduced
ram consumption.

On Sat, May 12, 2018 at 1:13 PM Gravemind <notifications@github.com> wrote:

> I could not get any call call to printf without adding at least 1KB of
> prog mem, so I went back to using Serial.print (at first to compare, then
> commit).
>
> Now, compared to f/configurable-scanners-and-debouncers:
>
>    - renamed debug_msg and print_result to debug_print and debug_resultln
>    - tweaked a couple of \n and messages
>    - removed the PSTR()/F() macros, so it uses ~40B less prog mem usage
>    but *adds ~500B of RAM* (*not sure if that's a good trade-off*, I can
>    put them back)
>    - reduced delay
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/keyboardio/avr_keyscanner/pull/9#issuecomment-388580082>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AACxaPD9k5z79n6cP3D1oUtdFXHVSEH1ks5tx0KFgaJpZM4T8Z9C>
> .
>

obra commented:
Have a look at the updates I pushed to configurable-scanners-and-debouncers
and tell me if they get close enough to what you were looking for? If so,
I'll resolve this.

On Sat, May 12, 2018 at 7:35 PM Jesse Vincent <jesse@keyboard.io> wrote:

> Thanks!
>
> Since we use progmem features elsewhere in the core, I suspect we're
> already paying that penalty, so I'd probably error on the side of reduced
> ram consumption.
>
> On Sat, May 12, 2018 at 1:13 PM Gravemind <notifications@github.com>
> wrote:
>
>> I could not get any call call to printf without adding at least 1KB of
>> prog mem, so I went back to using Serial.print (at first to compare,
>> then commit).
>>
>> Now, compared to f/configurable-scanners-and-debouncers:
>>
>>    - renamed debug_msg and print_result to debug_print and debug_resultln
>>    - tweaked a couple of \n and messages
>>    - removed the PSTR()/F() macros, so it uses ~40B less prog mem usage
>>    but *adds ~500B of RAM* (*not sure if that's a good trade-off*, I can
>>    put them back)
>>    - reduced delay
>>
>> —
>> You are receiving this because you commented.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/keyboardio/avr_keyscanner/pull/9#issuecomment-388580082>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/AACxaPD9k5z79n6cP3D1oUtdFXHVSEH1ks5tx0KFgaJpZM4T8Z9C>
>> .
>>
>

Gravemind commented:
Looking good, Thanks.

Don't know if I'm being too picky, but here is a last couple of new typo in the output:

```diff
diff --git a/etc/flash_firmware.ino b/etc/flash_firmware.ino
index 8cd96a6170..fe42c8c096 100644
--- a/etc/flash_firmware.ino
+++ b/etc/flash_firmware.ino
@@ -170,8 +170,9 @@ int write_firmware(uint8_t addr ) {
             data[data_counter++] = (0x00); // dummy end uint8_t
 
             result = twi_writeTo(addr, data, ELEMENTS(data), true, true);
-            debug_print(F(",Frame "));
+            debug_print(F("Frame "));
             debug_print(frame);
+            debug_print(F(" "));
             debug_print_result(result);
             // got something other than NACK. Start over.
             if (result != ENDTRANS_DATA_NACK) {
@@ -199,7 +200,7 @@ int verify_firmware(byte addr) {
         // skip the first 4 bytes, are they were probably overwritten by the reset vector preservation
         result = read_crc16(addr, &version, &crc16, offsets[0] + 4, firmware_length - 4);
 
-        debug_print(result);
+        debug_print_result(result);
 
         if (result != ENDTRANS_SUCCESS) {
             _delay_ms(100);
```

Closing the PR anyway. Thanks again.

obra commented:
Thanks. Merged.

On Sun, May 13, 2018 at 1:52 AM Gravemind <notifications@github.com> wrote:

> Closed #9 <https://github.com/keyboardio/avr_keyscanner/pull/9>.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/keyboardio/avr_keyscanner/pull/9#event-1623474723>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AACxaH0EdFFZ09JIggPKV4ihP6mQJYn2ks5tx_QxgaJpZM4T8Z9C>
> .
>
